ground/src/main/java/com/google/android/ground/ui/datacollection/TaskSequenceHandlerImpl.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection
+
+import com.google.android.ground.model.submission.TaskData
+import com.google.android.ground.model.task.Task
+
+class TaskSequenceHandlerImpl(
```

Comment:
If we do keep the interface, the distinguishing feature here, and the required data is the list of tasks, which must be non empty. You can probably better distinguish this class by naming it after that requirement, e.g.:

`NonEmptyTaskListSequenceHandler` or similar.

Reply:
Already replaced the interface with concrete class. Resolving this.